### ISSUE TITLE
PrefixLM didn't work because we didn't pass attention_mask correctly

### ISSUE THREAD

**Issue by thomasw21** [bigscience-workshop/Megatron-DeepSpeed#303]
Okay I have a strong intuition that the reason why we prefix LM never actually worked was because of this https://github.com/bigscience-workshop/Megatron-DeepSpeed/blob/e1c479e5890a4f38a743aabb31d1d95ff0e721f0/megatron/model/transformer.py#L588 Essentially `output` doesn't return the `attention_mask` and so effectively after the first layer that gets dropped, and so now the model thinks it has bidirectional views. I'm guessing that's why we saw rather low loss values.

cc @TevenLeScao

**Comment by thomasw21** [comment]
Nvm it should be working as expected. My bad for jumping the gun.
